src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Text.Json;
+using AzureExtension.Contracts;
+using DevHomeAzureExtension.DataModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+
+namespace AzureExtension.DevBox;
+
+public class DevBoxProvider : IComputeSystemProvider, IDisposable
+{
+    private readonly IHost _host;
+
+    public DevBoxProvider(IHost host)
+    {
+        _host = host;
+    }
+
+    string IComputeSystemProvider.DefaultComputeSystemProperties
+    {
+        get => throw new NotImplementedException();
+        set => throw new NotImplementedException();
+    }
+
+    public string DisplayName => "DevBox Provider";
+
+    public string Id => "182AF84F-D5E1-469C-9742-536EFEA94630";
+
+    public string Properties => throw new NotImplementedException();
+
+    public ComputeSystemProviderOperation SupportedOperations => throw new NotImplementedException();
+
+    private bool IsValid(JsonElement jsonElement)
+    {
+        return jsonElement.ValueKind != JsonValueKind.Undefined;
+    }
+
+    public IEnumerable<IComputeSystem> GetComputeSystemsAsync(IDeveloperId? developerId)
+    {
+        var computeSystems = new List<IComputeSystem>();
+
+        var mgmtSvc = _host.Services.GetService<IDevBoxManagementService>();

```

Comment:
Will this line crash when `GetComputeSystemAsync` is called from DevHome?  Will _host still be valid?

Reply:
Why would it crash? It can crash if all the service files are unavailable, which won't be the case in the actual implementation.